internal/services/mobilenetwork/mobile_network_slice_data_source.go

```diff
@@ -0,0 +1,138 @@
+package mobilenetwork
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/mobilenetwork/2022-11-01/mobilenetwork"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/mobilenetwork/2022-11-01/slice"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type SliceDataSource struct{}
+
+var _ sdk.DataSource = SliceDataSource{}
+
+func (r SliceDataSource) ResourceType() string {
+	return "azurerm_mobile_network_slice"
+}
+
+func (r SliceDataSource) ModelObject() interface{} {
+	return &SliceModel{}
+}
+
+func (r SliceDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return slice.ValidateSliceID
+}
+
+func (r SliceDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"mobile_network_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: mobilenetwork.ValidateMobileNetworkID,
+		},
+	}
+}
+
+func (r SliceDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"location": commonschema.LocationComputed(),
+
+		"single_network_slice_selection_assistance_information": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"slice_differentiator": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"slice_service_type": {
+						Type:     pluginsdk.TypeInt,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"tags": commonschema.TagsDataSource(),
+	}
+}
+
+func (r SliceDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var metaModel SliceModel
+			if err := metadata.Decode(&metaModel); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.MobileNetwork.SliceClient
+			mobileNetworkId, err := mobilenetwork.ParseMobileNetworkID(metaModel.MobileNetworkMobileNetworkId)
+			if err != nil {
+				return err
+			}
+
+			id := slice.NewSliceID(mobileNetworkId.SubscriptionId, mobileNetworkId.ResourceGroupName, mobileNetworkId.MobileNetworkName, metaModel.Name)
```

Comment:
Given that we are using the `mobile_network_id` and `name` here to build the SliceID, would it make sense to just have the datasource require just the SliceID to begin with? Then the MobileNetworkID could be parsed from that and we would only need one argument.

Reply:
My thought is `slice` is a nested resource inside `mobile network` like other resources inside `resource group`...

And with this, if it's better to require `resource_group_name`, `mobile_network_name` and `slice_name`?